### PR TITLE
Remove dependencies from tests/e2e/framework on pkg/scheduler/framework

### PR DESCRIPTION
**Pull request by alculquicondor**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Removes the transient dependencies from test/e2e/framework on pkg/scheduler/framework.

One of the removed dependency is
`test/e2e/framework` -> `pkg/kubelet/sysctl` -> `pkg/kubelet/lifecycle` -> `pkg/scheduler/framework/v1alpha1`, which was replaced by `test/e2e/framework` -> `pkg/kubelet/event`.

The other one goes from
`pkg/controller/daemon` -> `pkg/scheduler/framework/plugins/helper` -> `pkg/scheduler/framework/v1alpha1`. I removed the second import here. However, I noticed later that `test/e2e/framework` doesn't depend on `pkg/controller/daemon`, only on `pkg/controller`.

**Which issue(s) this PR fixes**:

Part of #89229

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by andrewsykim — COMMENTED** [review]
/approve
/lgtm

for test/e2e/framework

**Comment by alculquicondor** [comment]
and
/assign @ahg-g 
for scheduler

**Comment by ahg-g** [comment]
/lgtm

for the scheduler

File: pkg/kubelet/events/event.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ const (
 	PreemptContainer        = "Preempting"
 	BackOffStartContainer   = "BackOff"
 	ExceededGracePeriod     = "ExceededGracePeriod"
+	SysctlForbidden         = "SysctlForbidden"
```
Comment (mattjmcnaughton) [inline pkg/kubelet/events/event.go]: @alculquicondor , do you mind giving a short explanation of why we need to move around `pkg/kubelet` constants in order to support removing dependencies in `tests/e2e`? It looks like the change to stop depending on `pkg/kubelet/sysctl.go` is somewhat distinct? I'm not saying its a good or bad change, I'm just having a little trouble seeing how its related :)

**Review by mattjmcnaughton — COMMENTED** [review]
One high level question from me with a sig-node perspective :) Thanks for your work!

File: pkg/kubelet/events/event.go (hunk 1 above)
Comment (andrewsykim) [reply pkg/kubelet/events/event.go]: `pkg/kubelet/sysctl` depends on `pkg/kubelet/lifecycle` which depends on `pkg/scheduler/framework`.

Comment (alculquicondor) [reply pkg/kubelet/events/event.go]: the dependency chain goes like this:

`pkg/kubelet/sysctl` -> `pkg/kubelet/lifecycle` -> `pkg/scheduler/framework/...`

Comment (mattjmcnaughton) [reply pkg/kubelet/events/event.go]: Ahh I gotcha - mind including that information in the pull request or commit message?

Comment (alculquicondor) [reply pkg/kubelet/events/event.go]: lol, thanks @andrewsykim

File: pkg/kubelet/sysctl/whitelist.go [hunk 2]
```diff
@@ -22,14 +22,10 @@ import (
 
 	"k8s.io/kubernetes/pkg/apis/core/validation"
 	policyvalidation "k8s.io/kubernetes/pkg/apis/policy/validation"
+	"k8s.io/kubernetes/pkg/kubelet/events"
```
Comment (mattjmcnaughton) [inline pkg/kubelet/sysctl/whitelist.go]: For anyone else reading this... `pkg/kubelet/events` has no dependencies (is really just a list of constants). So there aren't any concerns about requiring it :)

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

LGTM from a sig-node perspective.

**Comment by andrewsykim** [comment]
Part of https://github.com/kubernetes/kubernetes/issues/74352

File: pkg/kubelet/events/event.go (hunk 1 above)
Comment (alculquicondor) [reply pkg/kubelet/events/event.go]: Updated the description

**Comment by alculquicondor** [comment]
All tests passing now

**Comment by alculquicondor** [comment]
@mattjmcnaughton anything else to add for approval?

**Comment by mattjmcnaughton** [comment]
> @mattjmcnaughton anything else to add for approvaLooking good! I think we'll want to rebase to resolve and conflict, and then get sign-off from one of the `pkg/kubelet` approvers - looks like @Random-Liu was tagged earlier.

**Comment by alculquicondor** [comment]
Conflict resolved

/unassign @Random-Liu 
/assign @tallclair

**Comment by alculquicondor** [comment]
After #89213, I'm not convinced of the importance of this PR.

Let me know if you disagree, but for now:
/close

**Pull request closed without merging**